fontes/avaliador-sintatico/dialetos/avaliador-sintatico-birl.ts

```diff
@@ -0,0 +1,99 @@
+import { Construto, FuncaoConstruto, Literal, Variavel } from '../../construtos';
+import { Escreva, Declaracao, Enquanto, Para, Escolha, Fazer } from '../../declaracoes';
+import { RetornoLexador, RetornoAvaliadorSintatico } from '../../interfaces/retornos';
+import { AvaliadorSintaticoBase } from '../avaliador-sintatico-base';
+import tiposDeSimbolos from '../../tipos-de-simbolos/birl';
+import { SimboloInterface } from '../../interfaces';
+import { ErroAvaliadorSintatico } from '../erro-avaliador-sintatico';
+
+export class AvaliadorSintaticoBirl extends AvaliadorSintaticoBase {
+    validarSegmentoHoraDoShow(): void {
+        this.consumir(tiposDeSimbolos.HORA_DO_SHOW, 'Esperado expressão `HORA DO SHOW` para iniciar o programa');
+        this.blocos += 1;
+    }
+
+    validarSegmentoBirlFinal(): void {
+        this.consumir(tiposDeSimbolos.BIRL, 'Esperado expressão `BIRL` para fechamento do programa');
+        this.blocos -= 1;
+    }
+
+    primario(): Construto {
+        // Creio q isso e para fazer a criação de variaveis
+        // mais não ha nenhum prefixo para criar variavel
+        // nessa linguagem
+        // acho q terei que pegar essa verificação com base no '='
+        // Em tese considerando que seja [ FRANGO FR = 'a'; ]
+        // new Variavel receberia no simbolo o valor de FR
+        if (this.verificarSeSimboloAtualEIgualA(tiposDeSimbolos.IGUAL)) {
```

Comment:
Pelo site, a criação da variável é pelo tipo de dados dela:

```
FRANGO FR = 'a'; (char)
MONSTRINHO M1 = 13; (short)
MONSTRO M2 = 37; (int)
MONSTRAO M3 = 666; (long)
TRAPEZIO T = 0.13; (float)
TRAPEZIO DESCENDENTE TD = 0.37; (double)
BICEPS FRANGO BF = 200; (unsigned)
```

Delégua usa apenas a palavra reservada `var` e infere o tipo. Aqui o programador precisa dizer o tipo, então as sete expressões pra tipo aqui atuam como `var`.

Não esqueça de remover o comentário.

Reply:
Entendi, Ok, removerei esses comentários e deixarei anotado essa observação para quando for implementar o avaliador sintático.